### PR TITLE
Fix boot order clearing

### PR DESCRIPTION
**Pull request by LIV2**
I was having an issue with the templates being built by packer having a Boot order set even though the cleanup of StepRun should have removed the bootorder by default.

The problem is that the boot order is cleared after the VM is marked as a template because it runs as a cleanup task for StepRun(), it doesn't throw an error but it doesn't do anything 
Even if you attempt the same with govc it just silently fails
```
# govc device.ls -vm /datacenter/vm/Templates/centos-7-270219-1102 -boot
cdrom-3000  VirtualCdrom  Remote device
# govc vm.markastemplate /datacenter/vm/Templates/centos-7-270219-1102
# govc device.boot -vm /datacenter/vm/Templates/centos-7-270219-1102 -order -
# govc device.ls -vm /datacenter/vm/Templates/centos-7-270219-1102 -boot
cdrom-3000  VirtualCdrom  Remote device
```

To resolve this problem I've simply moved setting/clearing the boot order into steps so that the clearing is done before the vm is marked as a template

### CODE REVIEW

File: common/step_setbootorder.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	vm := state.Get("vm").(*driver.VirtualMachine)
+
+	if s.BootOrder != "" {
+		ui.Say("Set boot order...")
+		order := strings.Split(s.BootOrder, ",")
+		if err := vm.SetBootOrder(order); err != nil {
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+	} else {
```
Comment (VladRassokhin) [inline common/step_setbootorder.go]: Since you're anyway editing this code, please convert into `else if s.SetOrder` :)

**Review by VladRassokhin — APPROVED** [review]
Looks great

File: common/step_setbootorder.go (hunk 1 above)
Comment (LIV2) [reply common/step_setbootorder.go]: Done! :)

**Comment by LIV2** [comment]
@mkuzmin Any chance of this still being merged?

**Comment by VladRassokhin** [comment]
This repo is deprecated. Please open PR in [official Packer repository](https://github.com/hashicorp/packer) if it's still relevant.

**Pull request closed without merging**